src/aspire/source/__init__.py

```diff
@@ -236,9 +306,32 @@ def vol_forward(self, vol, start, num):
         all_idx = np.arange(start, min(start + num, self.n))
         im = vol_project(vol, self.rots[all_idx, :, :])
 
-        im = self.filters(im, start, num)
+        im = self.eval_filters(im, start, num)
+
+        im = Image(im).shift(self.offsets[all_idx, :]).asnumpy()
 
-        im = im_translate(im, self.offsets[all_idx, :])
         im *= np.broadcast_to(self.amplitudes[all_idx], (self.L, self.L, len(all_idx)))
 
         return im
+
+    def to_starfile(self, mrcs_filename, block_name='', fields=None):
+        df = self._metadata.copy()
+        df['_image_name'] = pd.Series(['{0:05}@{1}'.format(i+1, mrcs_filename) for i in range(self.n)])
+        df = df.rename(self._metadata_aliases, axis=1)
```

Comment:
This is a non-issue now (in 836810a2ef2d85aeb930aee40e25a2152e68a642) since we no longer have an intermediate `StarfileStack` class. Aliases are applied for `RelionStarfileStack`, so one will never end up with a STAR file with non-relion fields. In other words, we only support a Relion STAR file, nothing else.